integration-libs/opf/cta/components/opf-cta-element/opf-cta-element.component.ts

```diff
@@ -5,23 +5,45 @@
  */
 
 import {
+  AfterViewInit,
   ChangeDetectionStrategy,
   Component,
   Input,
   inject,
 } from '@angular/core';
 import { DomSanitizer, SafeHtml } from '@angular/platform-browser';
+import { WindowRef } from '@spartacus/core';
+import { OpfDynamicScript } from '@spartacus/opf/base/root';
+import { OpfCtaScriptsService } from '../opf-cta-scripts/opf-cta-scripts.service';
 
 @Component({
   selector: 'cx-opf-cta-element',
   templateUrl: './opf-cta-element.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class OpfCtaElementComponent {
+export class OpfCtaElementComponent implements AfterViewInit {
   protected sanitizer = inject(DomSanitizer);
-  @Input() ctaScriptHtml: string;
+  protected opfCtaScriptsService = inject(OpfCtaScriptsService);
+  protected windowRef = inject(WindowRef);
 
+  @Input() ctaScriptHtml: OpfDynamicScript;
+
+  ngAfterViewInit(): void {
+    this.opfCtaScriptsService.loadAndRunScript(this.ctaScriptHtml);
+  }
   renderHtml(html: string): SafeHtml {
-    return this.sanitizer.bypassSecurityTrustHtml(html);
+    // Display sanitized html in SSR for security concerns
+    return this.windowRef.isBrowser()
+      ? this.sanitizer.bypassSecurityTrustHtml(this.removeScriptTags(html))
+      : html;
```

Comment:
[MEDIUM]
I'm having problem to understand what threat this line of code prevents in SSR. Is it _really_ needed?
Even if HTML is rendered into DOM, the NodeJS won't execute it. It's only the browser that will execute it. SSR with only output it as a big HTML string sent in the response to the browser.